examples/spotify-api-example/src/main/java/com/spotify/apollo/example/ArtistResource.java

```diff
@@ -0,0 +1,113 @@
+import com.spotify.apollo.Response;
+import com.spotify.apollo.Status;
+import com.spotify.apollo.example.data.Album;
+import com.spotify.apollo.example.data.Artist;
+import com.spotify.apollo.example.data.Track;
+import com.spotify.apollo.route.AsyncHandler;
+import com.spotify.apollo.route.Middlewares;
+import com.spotify.apollo.route.Route;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Stream;
+
+import okio.ByteString;
+
+/**
+ * The artist resource demonstrates how to use asynchronous routes in Apollo with query arguments.
+ */
+public class ArtistResource {
+
+  private static final String SPOTIFY_API = "https://api.spotify.com";
+  private static final String SEARCH_API = SPOTIFY_API + "/v1/search";
+  private static final String ARTIST_API = SPOTIFY_API + "/v1/artists";
+
+  public Stream<Route<AsyncHandler<Response<ByteString>>>> routes() {
+    // The artist resource has one parameterized route.
+    return Stream.of(
+        Route.async("GET", "/artists/toptracks/<country>", this::getArtistTopTracks)
+          .withMiddleware(Middlewares::autoSerialize)
+    );
+  }
+
+  public CompletionStage<Response<ArrayList<Track>>> getArtistTopTracks(RequestContext requestContext) {
+    // Validate request
+    Optional<String> query = requestContext.request().parameter("q");
+    if (!query.isPresent()) {
+      return CompletableFuture.completedFuture(
+          Response.forStatus(Status.BAD_REQUEST.withReasonPhrase("No search query")));
+    }
+    String country = requestContext.pathArgs().get("country");
+
+    // We need to first query the search API, parse the result, then query top-tracks.
+    Request searchRequest = Request.forUri(SEARCH_API + "?type=artist&q=" + query.get());
+    Client client = requestContext.requestScopedClient();
+    return client
+        .send(searchRequest)
+        .thenComposeAsync(response -> {
+          String topArtistId = parseFirstArtistId(response.payload().get().utf8());
+          Request topTracksRequest = Request.forUri(
+              String.format("%s/%s/top-tracks?country=%s", ARTIST_API, topArtistId, country));
+          return client.send(topTracksRequest);
+        }).thenApplyAsync(response -> Response.ok()
+            .withPayload(parseTopTracks(response.payload().get().utf8())))
+        .exceptionally(throwable -> Response
+            .forStatus(Status.INTERNAL_SERVER_ERROR.withReasonPhrase("Something went wrong")));
+  }
+
+  /**
+   * Parses an artist top tracks response from the
+   * <a href="https://developer.spotify.com/web-api/get-artists-top-tracks/">Spotify API</a>
+   *
+   * @param json The json response
+   * @return A list of top tracks
+   */
+  private ArrayList<Track> parseTopTracks(String json) {
+    ArrayList<Track> tracks = new ArrayList<>();
+    try {
+      ObjectMapper mapper = new ObjectMapper();
```

Comment:
ObjectMapper is thread safe and you don't need to make a new one every time :) Ie; the new call for each parsing is not needed. You can just have a static final instance somewhere that you reuse.